proxy/src/main/java/com/wavefront/agent/handlers/AbstractReportableEntityHandler.java

```diff
@@ -63,57 +64,54 @@
    * @param serializer           helper function to convert objects to string. Used when writing
    *                             blocked points to logs.
    * @param senderTasks          tasks actually handling data transfer to the Wavefront endpoint.
-   * @param setupMetrics         Whether we should report counter metrics.
+   * @param reportReceivedStats  Whether we should report a .received counter metric.
    * @param blockedItemsLogger   a {@link Logger} instance for blocked items
    */
   AbstractReportableEntityHandler(HandlerKey handlerKey,
                                   final int blockedItemsPerBatch,
                                   final Function<T, String> serializer,
                                   @Nullable final Collection<SenderTask<U>> senderTasks,
-                                  boolean setupMetrics,
+                                  boolean reportReceivedStats,
                                   @Nullable final Logger blockedItemsLogger) {
     this.handlerKey = handlerKey;
     //noinspection UnstableApiUsage
-    this.blockedItemsLimiter = blockedItemsPerBatch == 0 ? null :
-        RateLimiter.create(blockedItemsPerBatch / 10d);
+    this.blockedItemsLimiter = blockedItemsPerBatch == 0 ? null : RateLimiter.create(blockedItemsPerBatch / 10d);
     this.serializer = serializer;
     this.senderTasks = senderTasks == null ? new ArrayList<>() : new ArrayList<>(senderTasks);
+    this.reportReceivedStats = reportReceivedStats;
     this.rateUnit = handlerKey.getEntityType().getRateUnit();
     this.blockedItemsLogger = blockedItemsLogger;
 
-    MetricsRegistry registry = setupMetrics ? Metrics.defaultRegistry() : LOCAL_REGISTRY;
+    MetricsRegistry registry = reportReceivedStats ? Metrics.defaultRegistry() : LOCAL_REGISTRY;
     String metricPrefix = handlerKey.toString();
     MetricName receivedMetricName = new MetricName(metricPrefix, "", "received");
     MetricName deliveredMetricName = new MetricName(metricPrefix, "", "delivered");
     this.receivedCounter = registry.newCounter(receivedMetricName);
-    this.attemptedCounter = registry.newCounter(new MetricName(metricPrefix, "", "sent"));
+    this.attemptedCounter = Metrics.newCounter(new MetricName(metricPrefix, "", "sent"));
```

Comment:
should this be registry.newCounter() instead of Metrics.newCounter() ?

Reply:
@sushantdewan123 We want to always report .sent metric, but only report .received when reportReceivedCounter = true